interchain-token-factory/src/lib.rs

```diff
@@ -0,0 +1,332 @@
+#![no_std]
+
+use core::ops::Deref;
+
+use multiversx_sc::api::KECCAK256_RESULT_LEN;
+
+use token_manager::constants::{DeployTokenManagerParams, TokenManagerType};
+
+use crate::constants::{Hash, TokenId, PREFIX_CANONICAL_TOKEN_SALT, PREFIX_INTERCHAIN_TOKEN_SALT};
+use crate::proxy::CallbackProxy as _;
+
+multiversx_sc::imports!();
+
+pub mod constants;
+pub mod proxy;
+
+#[multiversx_sc::contract]
+pub trait InterchainTokenFactoryContract: proxy::ProxyModule {
+    #[init]
+    fn init(&self, interchain_token_service: ManagedAddress) {
+        require!(!interchain_token_service.is_zero(), "Zero address");
+
+        self.interchain_token_service()
+            .set_if_empty(&interchain_token_service);
+
+        self.chain_name_hash()
+            .set_if_empty(self.its_chain_name_hash());
+    }
+
+    // Needs to be payable because it issues ESDT token through the TokenManager
+    #[payable("*")]
+    #[endpoint(deployInterchainToken)]
+    fn deploy_interchain_token(
+        &self,
+        salt: Hash<Self::Api>,
+        name: ManagedBuffer,
+        symbol: ManagedBuffer,
+        decimals: u8,
+        initial_supply: BigUint,
+        minter: ManagedAddress,
+    ) -> TokenId<Self::Api> {
+        let sender = self.blockchain().get_caller();
+        let salt = self.interchain_token_salt(&self.chain_name_hash().get(), &sender, &salt);
+
+        let own_address = self.blockchain().get_sc_address();
+
+        let mut minter_bytes = &ManagedBuffer::new();
+
+        if initial_supply > 0 {
+            minter_bytes = own_address.as_managed_buffer()
+        } else if !minter.is_zero() {
+            minter_bytes = minter.as_managed_buffer()
+        };
```

Comment:
Yes, that is fine to not be handled. If `initial_supply = 0` then `minter_bytes` will remain an empty `ManagedBuffer`. This will trickle through the InterchainTokenService contract to the TokenManager contract with type MintBurn.
This means that no one else can mint this token besides the Interchain Token Service contract, which is perfectly fine.